### DISCUSSION TITLE
summarise not working

### DISCUSSION THREAD

**Discussion by AInisha** [pwwang/datar#183] [Announcements]
### datar version checks

- [X] I have checked that this issue has not already been reported.

- [ ] I have confirmed this bug exists on the **latest version** of datar and its backends.

### Issue Description

I am unable to use summarise(CNT= n()), i donot see a separate column with CN
```
dataframe= dataframe>>group_by(f.some_col1, f.some_col2)>>summarise(CNT= n())>>ungroup() >> arrange(f.some_col2)

dataframe = (dataframe.set_index('some_col2')
                                .groupby('some_col1', sort=False)
                                .resample('MS')['CNT'].max()
                                .unstack()
                               )
```
 error throws at .resample('MS')['CNT'].max(), as there is no CNT column exist, but earlier project ran fine for months.
Please assist me as early as possible on this
tried below 2 version as these are compatible with my project, earlier it was running fine:-
```
pip install -U datar==0.9.1
pip install -U datar[pdtypes]==0.9.1

pip install -U datar==0.9.0
pip install -U datar[pdtypes]==0.9.0
```
### Expected Behavior

the column CNT existed before

### Installed Versions

<details>

Replace this line with the output of datar.get_versions()
0.9.1
</details>

**Comment by pwwang** [comment]
Please:
1. try the latest version
2. provide a runnable example

**Comment by AInisha** [comment]
for the latest version i am not bale to use the datar at all
# !pip install -U datar==0.12.2
# !pip install -U datar[pdtypes]==0.12.2

data = {'Name': ['Tom', 'nick', 'krish', 'jack', 'jack'],
        'Age': [20, 21, 19, 18, 16]}
  
# Create DataFrame
df = pd.DataFrame(data)
df>>distinct(f.Name)


below is the error
ContextError: Cannot evaluate `ReferenceAttr` object without a context.

**Comment by AInisha** [comment]
our project is built with 
```
pip install -U datar==0.9.1
pip install -U datar[pdtypes]==0.9.1
```

now i used below:-
```
# !pip install -U datar==0.9.1
# !pip install -U datar[pdtypes]==0.9.1

data = {'Name': ['Tom', 'nick', 'krish', 'jack', 'jack'],
        'Age': [20, 21, 19, 18, 16]}
  
# Create DataFrame
df = pd.DataFrame(data)
df>>distinct(f.Name)
df>> group_by(f.Name)>>summarise(CNT= n())>>filter(CNT>0)
```
i see NameError: name 'CNT' is not defined

**Comment by AInisha** [comment]
there is no count column displays as separate, my point is that, earlier it used to do that

**Comment by AInisha** [comment]
is there any update

**Comment by pwwang** [comment]
![image](https://github.com/pwwang/datar/assets/1188067/f84eb3ec-7b8c-4cc6-b68a-b3a6e426f8fe)

Please follow the instructions above.

Note that the error you got `NameError: name 'CNT' is not defined` is because that you need `f.CNT`.

**Comment by AInisha** [comment]
let me try it with libraries you have used

**Comment by AInisha** [comment]
still getting similar  errors:
I am having  below as the output from get_versions, i think its due to version compatibility , Thank you for your help
python      : 3.8.10 
datar       : 0.12.2
simplug     : 0.3.2
executing   : 1.2.0
pipda       : 0.12.0
datar-numpy : 0.2.0
numpy       : 1.24.4
datar-pandas: 0.3.0
pandas      : 2.0.3

**Comment by pwwang** [comment]
Can you attach the full snapshot like I did, including the code and the error?

**Comment by pwwang** [comment]
I mean the full snapshot please, including the version printing, and module importing.

**Reply by AInisha** [reply]
I ran without downloading any additional library  and exceuted only latest datar and imported its library,  it ran fine gives separate column of count, showing the pandas version like -->pandas      : 1.2.4, python 3.8, other version remained same like your screenshot.
issue can be because our pandas version is  -->2.0.3, when executing with libraries,  some libraries which we need automatically takes  pandas 2.0.3 version i guess, is there any restriction to pandas version?

**Reply by pwwang** [reply]
Yes, pandas needs to be 1.x, pandas 2.x is a different story.

**Reply by AInisha** [reply]
Will there be any fix by this week so that datar latest version works with pandas >= 2.0.3

**Reply by pwwang** [reply]
No, pandas 2 uses pyarrow as backend. Quite a lot of stuff need to be reimplemented. But it's in the plan.

**Comment by GitHunter0** [comment]
> Will there be any fix by this week so that datar latest version works with pandas >= 2.0.3

@pwwang is incredibly efficient, but it's hard to keep up with so many changes. 
[dplyr 1.1.0](https://github.com/tidyverse/dplyr/releases/tag/v1.1.0), for example, received a lot of updates too.

**Reply by pwwang** [reply]
Thanks for bringing this. We are now aligning with dplyr 1.0.8
